chain/consensus/hierarchical/subnet/resolver/resolver.go

```diff
@@ -281,20 +292,25 @@ func (r *Resolver) HandleIncomingResolveMsg(ctx context.Context, sub *pubsub.Sub
 
 func (r *Resolver) processResolveMsg(ctx context.Context, submgr subnet.SubnetMgr, rmsg *ResolveMsg) (pubsub.ValidationResult, error) {
 	switch rmsg.Type {
-	case Push:
+	case PushMeta:
 		return r.processPush(ctx, rmsg)
 	case PullMeta:
-		return r.processPull(submgr, rmsg)
-	case Response:
-		return r.processResponse(ctx, rmsg)
+		return r.processPullMeta(submgr, rmsg)
+	case ResponseMeta:
+		return r.processResponseMeta(ctx, rmsg)
+	case PullLocked:
+		return r.processPullLocked(submgr, rmsg)
+	case ResponseLocked:
+		return r.processResponseLocked(ctx, rmsg)
```

Comment:
Should we add default section as guard?

Reply:
We implicitly have it. If no case is matched we return an error and a `ValidationReject`.